src/common/database.c

```diff
@@ -995,33 +996,25 @@ dt_database_t *dt_database_init(const char *alternative)
   lock_again:
     lock_tries++;
     old_mode = umask(0);
-    fd = open(db->lockfile, O_RDWR | O_CREAT | O_EXCL, 0666);
+    fd = fopen(db->lockfile, "r");
     umask(old_mode);
 
-    if(fd != -1) // the lockfile was successfully created - write our PID into it
+    if (fd)
     {
-      gchar *pid = g_strdup_printf("%d", getpid());
-      if(write(fd, pid, strlen(pid) + 1) > -1) db->lock_acquired = TRUE;
-      g_free(pid);
-      close(fd);
-    }
-    else // the lockfile already exists - see if it's a stale one left over from a crashed instance
-    {
-      char buf[64];
-      memset(buf, 0, sizeof(buf));
-      fd = open(db->lockfile, O_RDWR | O_CREAT, 0666);
-      if(fd != -1)
-      {
-        if(read(fd, buf, sizeof(buf) - 1) > -1)
+        pid_t pid;
+        if (fscanf(fd, "%d", &pid))
         {
-          int other_pid = atoi(buf);
-          if((kill(other_pid, 0) == -1) && errno == ESRCH)
+          if((kill(pid, 0) == -1) && errno == ESRCH)
           {
-            // the other process seems to no longer exist. unlink the .lock file and try again
-            unlink(db->lockfile);
+            // the other process seems to no longer exist. unlink the .lock file and try again            
             if(lock_tries < 5)
             {
-              close(fd);
+              int ret = fclose(fd);
+              if (ret) {
+                  fprintf(stderr, "[init] error opening the database lock file for reading\n");
+                  return NULL;
```

Comment:
Is this really what you want? The lockfile hasn't failed to open, it failed to close.